src/pages/registration/files_and_license_tab/FilesTableRow.tsx

```diff
@@ -0,0 +1,212 @@
+import { AssociatedFile, LicenseNames, licenses } from '../../../types/associatedArtifact.types';
+import { ConfirmDialog } from '../../../components/ConfirmDialog';
+import { SpecificFileFieldNames } from '../../../types/publicationFieldNames';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { TruncatableTypography } from '../../../components/TruncatableTypography';
+
+interface FilesTableRowProps {
+  file: AssociatedFile;
+  removeFile: () => void;
+  toggleLicenseModal: () => void;
+  baseFieldName: string;
+}
+
+export const FilesTableRow = ({ file, removeFile, baseFieldName }: FilesTableRowProps) => {
+  const { t } = useTranslation();
+  const [openConfirmDialog, setOpenConfirmDialog] = useState(false);
+  const toggleOpenConfirmDialog = () => setOpenConfirmDialog(!openConfirmDialog);
+  const { setFieldValue, setFieldTouched } = useFormikContext();
+
+  return (
+    <TableRow data-testid={dataTestId.registrationWizard.files.fileRow}>
+      <TableCell sx={{ wordBreak: 'break-word', minWidth: '8rem' }}>
+        <TruncatableTypography>{file.name}</TruncatableTypography>
+      </TableCell>
+
+      <TableCell align="center">
+        <Tooltip title={t('registration.files_and_license.remove_file')}>
+          <IconButton onClick={toggleOpenConfirmDialog}>
+            <CancelIcon color="error" />
+          </IconButton>
+        </Tooltip>
+        <ConfirmDialog
+          open={openConfirmDialog}
+          title={t('registration.files_and_license.remove_file')}
+          onAccept={() => {
+            removeFile();
+            toggleOpenConfirmDialog();
+          }}
+          onCancel={toggleOpenConfirmDialog}>
+          <Typography>
+            {t('registration.files_and_license.remove_file_description', { fileName: file.name })}
+          </Typography>
+        </ConfirmDialog>
+      </TableCell>
+
+      <TableCell align="center">{prettyBytes(file.size)}</TableCell>
+
+      <TableCell align="center">
+        <Field name={`${baseFieldName}.${SpecificFileFieldNames.AdministrativeAgreement}`}>
+          {({ field }: FieldProps) => (
+            <FormControlLabel
+              data-testid={dataTestId.registrationWizard.files.administrativeAgreement}
+              control={
+                <Tooltip title={t('registration.files_and_license.administrative_contract')}>
+                  <Checkbox
+                    {...field}
+                    checked={field.value}
+                    onChange={(event) => {
+                      field.onChange(event);
+                      setFieldValue(`${baseFieldName}.${SpecificFileFieldNames.PublisherAuthority}`, null);
+                      setFieldValue(`${baseFieldName}.${SpecificFileFieldNames.License}`, null);
+                      setFieldValue(`${baseFieldName}.${SpecificFileFieldNames.EmbargoDate}`, null);
+                    }}
+                  />
+                </Tooltip>
+              }
+              label=""
```

Comment:
Hvorfor tom label? 🤔

Reply:
Teksten ligger nå i tooltip. Burde jeg fikse en aria-labeledby her?

Reply:
aha. Ser vi håndterer dette litt ulikt på forskjellige plasser fra før, enten ved å sette title i Tootip, eller ved å sette label på FormControlLabel (uten å bruke Tooltip)
Det enkleste er kanskje å bruke label til FormControlLabel uten Tooltip..?

Reply:
Fikser i annen PR, da jeg uansett må rote litt her for å fikse bug med UnpublishableFile